test/integration/homepage_popular_links_test.rb

```diff
@@ -133,6 +133,18 @@ class HomepagePopularLinksTest < JavascriptIntegrationTest
 
       assert_not page.has_text?("www.abc.com")
     end
+
+    should "display 'service problem' error message when publishing api gives an error" do
```

Comment:
Do we need the extra tests in here? Don't they just duplicate the ones in `test/functional/homepage_controller_test.rb`? I think the integration tests are slower, so we should try and not have too many of them.

If we do want to keep them, we should probably extract the stubbing to methods like is done for the 
tests in `test/functional/homepage_controller_test.rb`.

Reply:
Yeah, we can get rid of these now.